packages/contracts-bridge/contracts/test/TokenRegistry.t.sol

```diff
@@ -1,10 +1,548 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 pragma solidity 0.7.6;
 
+// Local imports
 import {BridgeTest} from "./utils/BridgeTest.sol";
+import {BridgeMessage} from "../BridgeMessage.sol";
+import {BridgeToken} from "../BridgeToken.sol";
+import {Encoding} from "../Encoding.sol";
+
+// External imports
+import {TypeCasts} from "@nomad-xyz/contracts-core/contracts/XAppConnectionManager.sol";
+import {TypedMemView} from "@summa-tx/memview-sol/contracts/TypedMemView.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 contract TokenRegistryTest is BridgeTest {
+    using TypedMemView for bytes;
+    using TypedMemView for bytes29;
+    using BridgeMessage for bytes29;
+
     function setUp() public override {
         super.setUp();
     }
+
+    function test_getCanonicalTokenId() public {
+        (uint32 domain, bytes32 id) = tokenRegistry.getCanonicalTokenId(
+            remoteTokenLocalAddress
+        );
+        assertEq(uint256(domain), remoteDomain);
+        assertEq(id, remoteTokenRemoteAddress);
+    }
+
+    function test_getCanonicalTokenIdFuzzed(
+        uint32 fuzzedDomain,
+        bytes32 fuzzedId
+    ) public {
+        address localAddress = createRemoteToken(fuzzedDomain, fuzzedId);
+        (uint32 domain, bytes32 id) = tokenRegistry.getCanonicalTokenId(
+            localAddress
+        );
+        assertEq(uint256(domain), fuzzedDomain);
+        assertEq(id, fuzzedId);
+    }
+
+    function test_getRepresentationAddress() public {
+        address repr = tokenRegistry.getRepresentationAddress(
```

Comment:
consider: 
- random domain, random id 
- `remoteDomain`, random id
- random domain, `remoteTokenRemoteAddress`

should all return address(0)